Header header
bool[] gate_open
bool[] charge_field_occupied